src/main/java/com/powsybl/nad/build/iidm/NetworkGraphBuilder.java

```diff
@@ -70,119 +72,125 @@ private VoltageLevelNode createVoltageLevelNode(VoltageLevel vl, Graph graph, bo
         return vlNode;
     }
 
-    private class VisitorBuilder extends DefaultTopologyVisitor {
-        private final Graph graph;
+    private void visitEquipments(VoltageLevel vl, Graph graph) {
+        vl.getLineStream().forEach(l -> visitLine(vl, l, graph));
+        vl.getTwoWindingsTransformerStream().forEach(twt -> visitTwoWindingsTransformer(vl, twt, graph));
+        vl.getThreeWindingsTransformerStream().forEach(thwt -> visitThreeWindingsTransformer(vl, thwt, graph));
+        vl.getConnectableStream(HvdcConverterStation.class).forEach(hvdc -> visitHvdcConverterStation(hvdc, graph));
+    }
 
-        public VisitorBuilder(Graph graph) {
-            this.graph = graph;
-        }
+    private void visitLine(VoltageLevel vl, Line line, Graph graph) {
+        Branch.Side side = line.getTerminal(Branch.Side.ONE).getVoltageLevel() == vl ? Branch.Side.ONE : Branch.Side.TWO;
```

Comment:
To avoid duplicated code this can be done in addEdge, as it's the same for two-winding transformers.